### PR TITLE
ci(mergify): update commit message template

### LINKED ISSUE #6791: chore: Mergify should replace squashed log messages with message from PR description
### Is your feature request related to a problem? Please describe.

Currently we get logs like this when merging with mergify. 

image

In the PR description we could have a section called PR message and mergify could get that and use it. This way you can still set your custom message.

https://docs.mergify.com/configuration/#data-type-template

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Update Mergify's commit message template to use the "What's changed and what's your intention?" section in the PR description as the message of the squashed commit.

This change has been made by @fuyufjh from Mergify config editor.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

None

## Refer to a related PR or issue link (optional)

closes #6791

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -1,3 +1,13 @@
+shared:
+  - commit_message_template: &commit_message_template |
+      {{title}}
+      
+      {{body}}
```
Comment (xxchan) [inline .mergify.yml:5]: Is this full PR description? I think current PR template is noisy and is not suitable for commit message 🥵

Comment (fuyufjh) [reply .mergify.yml:5]: Yeah a little bit nosiy. May discuss here: 6791

Comment (CAJan93) [reply .mergify.yml:5]: Adding comment in https://github.com/risingwavelabs/risingwave/issues/6791

File: .mergify.yml [hunk 2]
```diff
@@ -1,3 +1,13 @@
+shared:
+  - commit_message_template: &commit_message_template |
+      {{title}}
+      
+      {{body | get_section("## What's changed and what's your intention?")}}
+      
+      {% for user in approved_reviews_by %}
+      Approved-By: {{user}}
+      {% endfor %}
+
```
Comment (xxchan) [inline .mergify.yml:10]:
```suggestion

{% for commit in commits %}
Co-Authored-By: {{ commit.author }} <{{ commit.email_author }}>
{% endfor %}
```

🤔

(But maybe also need to dedup (and exclude bot?)...

File: .mergify.yml [hunk 3]
```diff
@@ -1,3 +1,13 @@
+shared:
+  - commit_message_template: &commit_message_template |
+      {{title}}
+      
+      {{body | get_section("## What's changed and what's your intention?")}}
```
Comment (xxchan) [inline .mergify.yml:5]: Considering many ppl don't keep the section very clean, it's still a little noisy. But maybe acceptable. 😇

Comment (fuyufjh) [reply .mergify.yml:5]: It's another reason to urge us to write a cleaner "What's changed" description. Let me announce it again in public channels 🤣

File: .mergify.yml (hunk 2 above)
Comment (TennyZhuang) [reply .mergify.yml:10]: I guess you need [unique filter](https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.unique)?

File: .mergify.yml (hunk 3 above)
Comment (TennyZhuang) [reply .mergify.yml:5]: What'll happen if we paste an image here?

File: .mergify.yml (hunk 2 above)
Comment (CAJan93) [reply .mergify.yml:10]: Won't this result in 

```
Co-Authored-By: a a.b@mail.com
Co-Authored-By: b c.d@mail.com
Co-Authored-By: c d.e@mail.com
```

Don't we want 

```
Co-Authored-By: 
{% for commit in commits %}
{{ commit.author }} <{{ commit.email_author }}>
{% endfor %}
```

File: .mergify.yml (hunk 3 above)
Comment (CAJan93) [reply .mergify.yml:5]: Good question @TennyZhuang. We should definitely try this before merging this PR

File: .mergify.yml (hunk 2 above)
Comment (xxchan) [reply .mergify.yml:10]: image

Comment (CAJan93) [reply .mergify.yml:10]: Okay. If that is the layout we want then this is good 👍

Comment (fuyufjh) [reply .mergify.yml:14]: Updated. Please take a look

File: .mergify.yml (hunk 3 above)
Comment (fuyufjh) [reply .mergify.yml:5]: Perhaps you can only try it after this PR being merged 😇

**Review by CAJan93 — APPROVED** [review]
Let's merge it and give it a try?

**Comment by BugenZhao** [comment]
The PR number in title seems missing.

image

**Comment by fuyufjh** [comment]
> The PR number in title seems missing.

Let me fix it ASAP.

**Pull request merged**